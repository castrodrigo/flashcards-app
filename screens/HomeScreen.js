import React from "react";
import { ScrollView, Text } from "react-native";
import styled from "styled-components/native";

const Container = styled.View`
  flex: 1;
  padding-top: 30px;
`;

export default class HomeScreen extends React.Component {
  static navigationOptions = {
    header: null
  };

  render() {
    return (
      <Container>
        <ScrollView>
          <Text>Home</Text>
        </ScrollView>
      </Container>
    );
  }
}
