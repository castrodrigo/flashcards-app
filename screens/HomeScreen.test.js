import React from "react";
import { shallow } from "enzyme";
import HomeScreen from "./HomeScreen";

const props = {
  decksMap: ["321"],
  decks: { "321": { id: "321", name: "Deck Test" } },
  cards: { "321": { "123": { name: "Question", answer: "Answer" } } },
  navigation: { navigate: jest.fn() }
};

describe("Render HomeScreen", () => {
  it("should render successfully", () => {
    const wrapper = shallow(<HomeScreen {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
