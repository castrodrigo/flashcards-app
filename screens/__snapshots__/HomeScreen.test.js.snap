// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HomeScreen should render successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeScreen
    cards={
      Object {
        "321": Object {
          "123": Object {
            "answer": "Answer",
            "name": "Question",
          },
        },
      }
    }
    decks={
      Object {
        "321": Object {
          "id": "321",
          "name": "Deck Test",
        },
      }
    }
    decksMap={
      Array [
        "321",
      ]
    }
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <DeckListItem
          cardsCount={1}
          deck={
            Object {
              "id": "321",
              "name": "Deck Test",
            }
          }
          navigation={
            Object {
              "navigate": [MockFunction],
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "321",
        "nodeType": "function",
        "props": Object {
          "cardsCount": 1,
          "deck": Object {
            "id": "321",
            "name": "Deck Test",
          },
          "navigation": Object {
            "navigate": [MockFunction],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <DeckListItem
            cardsCount={1}
            deck={
              Object {
                "id": "321",
                "name": "Deck Test",
              }
            }
            navigation={
              Object {
                "navigate": [MockFunction],
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "321",
          "nodeType": "function",
          "props": Object {
            "cardsCount": 1,
            "deck": Object {
              "id": "321",
              "name": "Deck Test",
            },
            "navigation": Object {
              "navigate": [MockFunction],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
